### DISCUSSION TITLE
Flunt 2.0.5

### DISCUSSION THREAD

**Discussion by rafaferreirasantos** [balta-io/7196#37] [Perguntas e Respostas] [answered]
Bom dia, alguém, chegou a implementar o ICommand utilizando a versão 2.0.5 do flunt?

Como ficaria? Vi que não temos mais a interface IValidatable apenas INotifiable.

Utilizei da seguinte forma:
Deixei de implementar a interface na ICommand,  usei direto no CreateTodoCommand
É isso mesmo?

```
using Flunt.Notifications;
using Flunt.Validations;
using TodoApp.Domain.Commands.Contracts;

namespace TodoApp.Domain.Commands;
public class CreateTodoCommand : Notifiable<Notification>, ICommand
{
    public string Title { get; set; }
    public DateTime Date { get; set; }
    public string User { get; set; }

    public CreateTodoCommand(string title, DateTime date, string user)
    {
        Title = title;
        Date = date;
        User = user;
    }

    public void Validate()
    {
        AddNotifications(
            new Contract<string>()
                .Requires()
                .IsGreaterThan(Title, 3, "Title", "O título do ToDo deve conter pelo menos 3 caracteres.")
                .IsGreaterThan(User, 6, "User", "Usuário inválido.")
        );
    }
}
```

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @zecaloteiro, a recomendação é usar a mesma versão do curso, mas aqui nas docs do Flunt tem os exemplos:
https://github.com/andrebaltieri/Flunt

Mais precisamente aqui:
https://github.com/andrebaltieri/Flunt/wiki/03.-Notifications

**Comment by llmeireles** [comment]
Boa tarde, @andrebaltieri acredito que o @zecaloteiro , esteja falando da aula "Definindo o contrato dos comandos" do curso "Todo App: API com ASP.NET Core, CQRS e EF Core", onde é utilizado o pacote flunt , para fazer as validações. Consultando os pacotes encontrei também o pacote do @andrebaltieri  https://github.com/andrebaltieri/FluentValidator/tree/cffe8e1d39b453d0a9a50a6b2398d0ba71fd1d9c, que trata exatamente dessa interface IValidatable.

**Reply by andrebaltieri** [reply]
@zecaloteiro @llmeireles dependendo do curso eu usei versão diferente do Flunt, então recomendo utilizarem as mesmas versões:
`dotnet add package flunt --version 1.0.5` por exemplo....

Caso queiram atualizar para a versão mais nova, ai precisa olhar no GitHub do projeto para ver as mudanças... a classe `IValidatable` não existe mais, mas vocês podem criar ela no projeto de vocês caso queiram... Tem registro dela nos códigos antigos do Flunt.

**Reply by llmeireles** [reply]
Perfeito @andrebaltieri , obrigado pelo retorno!
